src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -307,7 +307,15 @@ const buildWhereClause = ({ schema, query, index }) => {
     }
 
     if (Array.isArray(fieldValue.$all) && isArrayField) {
-      patterns.push(`array_contains_all($${index}:name, $${index + 1}::jsonb)`);
+      if (fieldValue.$all[0].$regex) {
```

Comment:
Here I just checked the first index but you can loop through and check all fields instead of checking on the controller